### PR TITLE
Feature/session timeout

### PR DESCRIPTION
**Pull request by erocknine**
https://trello.com/c/3MDz7gnh/32-session-timeout

### Timeout Component
- Added Timeout component in /components/Timeout. Timeout component wraps entire application in App.js.
- Timeout component uses .env file with variables `REACT_APP_IDLETIME` and `REACT_APP_COUNTDOWNTIME` representing time in milliseconds.
- `idleTime` determines the duration of time a user is idle before launching countdown modal. `countdownTime` determines the duration before the user is automatically logged off, duration displayed on UI modal to user.

### Countdown Modal Component
- Added countdown timer modal containing log off and continue buttons. Countdown time is shown to users.
- Changed previous CSS to css-in-js using Material.
- Created storybook and unit test

## To Dos
- Possibly use JS Date() to check idle time for more accuracy.

## Screenshot
Screenshot_2021-01-11 Femmecubator
Screenshot_2021-01-11 Femmecubator(2)

Co-authored by: @sherouk

### CODE REVIEW

File: src/components/Timeout/Timeout.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    background: 'rgb(56, 132, 255)',
+    marginLeft: '1rem',
+    '&:hover': {
+      filter: 'brightness(1.2)',
+      transition: '300ms ease-in-out',
+    },
+  },
+});
+
+export default function Timeout(props) {
```
Comment (cfernando415) [inline src/components/Timeout/Timeout.js:92]: lets destructure the props:

`export default function Timeout({ children }) {...`

File: src/components/Timeout/Timeout.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      {idle && (
+        <Modal
+          countdownTime={countdownTime}
+          setIdle={setIdle}
+          logoff={logoff}
+          timedOut={timedOut}
+        />
+      )}
+      {props.children}
+    </div>
```
Comment (cfernando415) [inline src/components/Timeout/Timeout.js:135]: refactor, if you have more than 2 or more attribs use es6 syntax:

`<div onMouseOver={reset} onKeyPress={reset}>
      {idle && (
        <Modal
          {...{countdownTime: countdownTime,
          setIdle: setIdle,
          logoff: logoff,
          timedOut: timedOut}}
        />)}{children}</div>`

File: src/components/Timeout/Timeout.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                {`${minutes} min ${seconds < 10 ? '0' : ''}${seconds} secs`}
+              </p>
+              <p className={classes.modal__center_div__content__p_text}>
+                Choose continue to keep working or log off.
+              </p>
+            </div>
+            <div className={classes.modal__center_div__buttons}>
+              <span
+                className={classes.modal__center_div__buttons__button_logoff}
+                onClick={() => logoff()}
```
Comment (cfernando415) [inline src/components/Timeout/Timeout.js:187]: `onClick={logoff}`

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please see comments

File: src/components/Timeout/ModalPopup.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                Choose continue to keep working or log off.
+              </p>
+            </div>
+            <div className={classes.modal__center_div__buttons}>
+              <Button
+                className={classes.modal__center_div__buttons__button_logoff}
+                onClick={logoff}
+                onKeyPress={logoff}
+                aria-label="log off"
+                tabIndex="0"
```
Comment (cfernando415) [inline src/components/Timeout/ModalPopup.js:56]: let's rewrite using es6
`{...{className: classes.modal__center_div__buttons__button_logoff,
                onClick: logoff,
                onKeyPress: logoff,
                'aria-label': 'log off'
                tabIndex: '0'}}`

File: src/components/Timeout/ModalPopup.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+              >
+                Log Off
+              </Button>
+              <Button
+                className={classes.modal__center_div__buttons__button_continue}
+                onClick={() => setIdle(false)}
+                onKeyPress={() => setIdle(false)}
+                aria-label="continue"
+                tabIndex="0"
```
Comment (cfernando415) [inline src/components/Timeout/ModalPopup.js:65]: same here

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see comments

File: package.json [hunk 6]
```diff
@@ -92,8 +93,7 @@
     "react-test-renderer": "^17.0.2"
   },
   "prettier": {
-    "singleQuote": true,
-    "arrowParens": "avoid"
+    "singleQuote": true
```
[the remark this answers is not in the document]
Comment (sherouk) [reply package.json:96]: good question...not sure. will look into it

**Comment by sherouk** [comment]
@cfernando415 requested changes complete

**Review by cfernando415 — APPROVED** [review]
:+1:

**Pull request merged**